unittest/test_ccache.cpp

```diff
@@ -191,4 +191,82 @@ TEST_CASE("rewrite_dep_file_paths")
   }
 }
 
+TEST_CASE("parse_depfile")
+{
+  SUBCASE("Parse empty depfile")
+  {
+    std::vector<std::string> result = parse_depfile("");
+    CHECK(result.size() == 0);
+  }
+
+  SUBCASE("Parse simple depfile")
+  {
+    std::vector<std::string> result = parse_depfile("cat.o: meow meow purr");
+    CHECK(result.size() == 4);
+    CHECK(result[0] == "cat.o:");
+    CHECK(result[1] == "meow");
+    CHECK(result[2] == "meow");
+    CHECK(result[3] == "purr");
+  }
+
+  SUBCASE("Parse depfile with a backslash newline")
+  {
+    std::vector<std::string> result =
+      parse_depfile("cat.o: meow\\\nmeow\\\n purr");
+    CHECK(result.size() == 3);
+    CHECK(result[0] == "cat.o:");
+    CHECK(result[1] == "meowmeow");
+    CHECK(result[2] == "purr");
+  }
```

Comment:
This behavior does not match what I see with GNU Make 4.2.1. Do you run another Make version?

A Makefile like

```
cat.o: meow\
meow\
 purr
```

will be interpreted as if it looked like

```
cat.o: meow meow purr
```

i.e. without joining the tokens separated by backslash + newline.

Reply:
So, looks like this leading prefix rule only applied to the recipe part, not prerequisite part and `\\\n` can be simply replaced as `\n` for this purpose.